### PR TITLE
Added BookStack

### PR DESCRIPTION
**Pull request by tfyl**
## Purpose

Enable BookStack in DockSTARTer which is 'A platform for storing and organising information and documentation'

## Approach

I added necessary files to enable BookStack in DockSTARTer

#### Learning

First, I set up BookStack to test out as a `.override` in DS then I transitioned to converting this into a DS compatible layout (i.e the different files needed in `.apps`). I used the [LSIO documentation](https://github.com/linuxserver/docker-bookstack) to understand the container and the different environment variables. For more information, you can visit the [BookStack website](https://www.bookstackapp.com/) and their subsequential [GitHub repository](https://github.com/BookStackApp/BookStack).

https://github.com/linuxserver/docker-bookstack
https://www.bookstackapp.com/
https://github.com/BookStackApp/BookStack

## Requirements

- [x] These changes meet the standards for [contributing](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CONTRIBUTING.md).
- [x] I have read the [code of conduct](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: compose/.apps/bookstack/bookstack.ports.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+services:
+  bookstack:
+    ports:
+      - "${BOOKSTACK_PORT_6875}:80"
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.ports.yml]: Use `BOOKSTACK_PORT_80` for the variable. The default value of the variable in .env can still be 6875

File: compose/.apps/bookstack/bookstack.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+services:
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Missing
```
# APPNICENAME=
# APPDESCRIPTION=
```

File: compose/.apps/bookstack/bookstack.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  bookstack:
+    image: linuxserver/bookstack
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Image does not go here. It is in the architecture specific files.

File: compose/.apps/bookstack/bookstack.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    container_name: bookstack
+    restart: unless-stopped
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Missing logging (copy paste from any other template).

File: compose/.apps/bookstack/bookstack.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    environment:
+      - PUID=${PUID}
+      - PGID=${PGID}
+      - DB_HOST=${BOOKSTACK_DB_HOST}
+      - DB_USER=${BOOKSTACK_DB_USER}
+      - DB_PASS=${BOOKSTACK_DB_PASS}
+      - DB_DATABASE=${BOOKSTACK_DB_DATABASE}
+      - APP_URL=${BOOKSTACK_URL}
+    volumes:
+      - ${DOCKERCONFDIR}/bookstack:/config
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Missing localtime and shared volumes.

File: compose/.apps/bookstack/bookstack.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+services:
+  bookstack:
+    image: linuxserver/bookstack
+    container_name: bookstack
+    restart: unless-stopped
+    environment:
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Missing TZ variable. Also sort these alphabetically.

File: compose/.env.example [hunk 7]
```diff
@@ -87,6 +87,16 @@ BOOKSONIC_NETWORK_MODE=
...
 
+### BOOKSTACK
+BOOKSTACK_ENABLED=false
+BOOKSTACK_BACKUP_CONFIG=true
+BOOKSTACK_DB_HOST=
+BOOKSTACK_DB_USER=
+BOOKSTACK_DB_PASS=
+BOOKSTACK_DB_DATABASE=
+BOOKSTACK_URL=
+BOOKSTACK_PORT_6875=6875
```
Comment (nemchik) [inline compose/.env.example]: Ports go under BACKUP_CONFIG. The rest should be alphabetical beneath that.

File: compose/.apps/bookstack/bookstack.yaml (hunk 5 above)
Comment (nemchik) [reply compose/.apps/bookstack/bookstack.yaml]: Please sort these (always sort :smile: )

File: compose/.env.example [hunk 8]
```diff
@@ -87,6 +87,17 @@ BOOKSONIC_NETWORK_MODE=
...
 
+### BOOKSTACK
+BOOKSTACK_ENABLED=false
+BOOKSTACK_BACKUP_CONFIG=true
+BOOKSTACK_PORT_80=6875
+BOOKSTACK_DB_DATABASE=
+BOOKSTACK_DB_HOST=
+BOOKSTACK_DB_PASS=
+BOOKSTACK_URL=
+BOOKSTACK_DB_USER=
```
Comment (nemchik) [inline compose/.env.example]: I think URL comes after DB_USER

File: compose/.apps/bookstack/bookstack.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    logging:
+      driver: "json-file"
+      options:
+        max-file: ${DOCKERLOGGING_MAXFILE}
+        max-size: ${DOCKERLOGGING_MAXSIZE}   
+    restart: unless-stopped   
+    volumes:
+      - ${DOCKERCONFDIR}/bookstack:/config
+      - /etc/localtime:/etc/localtime:ro
+      - ${DOCKERSHAREDDIR}:/shared
```
Comment (nemchik) [inline compose/.apps/bookstack/bookstack.yaml]: Also this is missing the empty line at the end of the file which caused the build to fail https://travis-ci.com/GhostWriters/DockSTARTer/jobs/203761717

File: compose/.apps/bookstack/bookstack.yaml (hunk 5 above)
Comment (tfyl) [reply compose/.apps/bookstack/bookstack.yaml]: These were sorted – weren't they.

Comment (nemchik) [reply compose/.apps/bookstack/bookstack.yaml]: image
This has P, D, A. Should be A, D, P.

**Pull request merged**